src/main/kotlin/rys/ajaxpetproject/service/impl/UserServiceImplementation.kt

```diff
@@ -0,0 +1,39 @@
+package rys.ajaxpetproject.service.impl
+
+import org.bson.types.ObjectId
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Service
+import rys.ajaxpetproject.exception.UserNotFoundException
+import rys.ajaxpetproject.model.MongoUser
+import rys.ajaxpetproject.repository.UserRepository
+import rys.ajaxpetproject.service.UserService
+
+@Service
+class UserServiceImplementation(
+    val userRepository: UserRepository,
+    val passwordEncoder: PasswordEncoder) : UserService {
+
+    override fun createUser(mongoUser: MongoUser): MongoUser {
+        return userRepository.save(mongoUser.copy(password = passwordEncoder.encode(mongoUser.password)))
+    }
+
+    fun getUserById(id: ObjectId): MongoUser = userRepository.findUserById(id) ?: throw UserNotFoundException()
+
+    override fun findUserById(id: ObjectId): MongoUser? = userRepository.findUserById(id)
+
+    override fun findAllUsers(): List<MongoUser>? = userRepository.findAllBy()
```

Comment:
is it possible to have a null result here? fix in all similar places

Reply:
<img width="477" alt="image" src="https://github.com/BohdanRysakov/AjaxOnBoardRysakov/assets/37266488/dc03a04e-f357-4230-8c26-cfa05a676827">
Possible

Reply:
but it's another method) I'm talking about findAll...)